### ISSUE TITLE
I will bet one beer...

### ISSUE THREAD

**Issue by timholy** [andyferris/Traitor.jl#8]
...that this can be implemented now. While memory is hazy, I *think* the missing ingredient before was the ability to redefine the "trait dispatch depot" function and its recipients with new signatures when you extend this list of traits. But now we have `Base.delete_method` :smile:.

**Comment by andyferris** [comment]
Yes - I was thinking the same. :)

**Comment by timholy** [comment]
Glad to know you're as evil as I am. :imp: 

Wanna talk about this a bit at the hackathon? CC @mauro3.

**Comment by s-celles** [comment]
Hello,

What is current status of Traitor and Julia 0.7/1.0?
I need to port a C++ code with multiple inheritance and traits may be a way to do it (I'm also considering composition over inheritance but it will probably force me to duplicate some functions

Kind regards

**Comment by timholy** [comment]
Right now it's a playground for ambitious developers who want to make a major contribution to Julia by implementing it :smile:. Feel free to help out!

**Comment by andyferris** [comment]
To expand on that, it hasn’t been updated for the recent versions of Julia, and even on older versions it was incomplete in terms of reaching its goals. We would love to make it so but as a playground project it gets a lower priority on development time. 

You can always perform trait based dispatch with your own dispatch pattern in order to port your code. We use such patterns throughout Julia and it’s ecosystem.

**Comment by mauro3** [comment]
You can use the registered https://github.com/mauro3/SimpleTraits.jl, if you don't want to hand-code your trait-dispatch (with the added bonus of clearly marking traits for future refactors).  Note though that the supported feature-set is less than what you can hand-code.

**Comment by c42f** [comment]
Agreed, SimpleTraits is a much better package to look at if you're trying to get "real work" done right now, as opposed to having fun playing compiler/language developer :-P

**Comment by MasonProtter** [comment]
> ...that this can be implemented now. 

Sorry could someone clarify what "this" refers to here?

**Comment by andyferris** [comment]
> Sorry could someone clarify what "this" refers to here?

This package, or else the idea that this package represents? I haven't read it yet, but I'm supposing your #10?

**Comment by MasonProtter** [comment]
Oh, I see now. I didn't realize the issue was opened *after* this package stopped working. I thought Tim was talking about some specific feature :facepalm:

**Comment by andyferris** [comment]
It was, in fact, born stillborn - around the same time as the version of Julia that “broke” it.

It’s not really functional without `add_backedge!` which I only learned about this week.
